src/map.c

```diff
@@ -26,14 +28,17 @@ void check_vector(SEXP x, const char *name) {
 }
 
 // call must involve i
-SEXP call_loop(SEXP env, SEXP call, int n, SEXPTYPE type, int force_args) {
+SEXP call_loop(SEXP env, SEXP call, int n, SEXPTYPE type, int force_args,
+               SEXP progress) {
   // Create variable "i" and map to scalar integer
   SEXP i_val = PROTECT(Rf_ScalarInteger(1));
   SEXP i = Rf_install("i");
   Rf_defineVar(i, i_val, env);
 
+  SEXP bar = PROTECT(cli_progress_bar(n, progress));
   SEXP out = PROTECT(Rf_allocVector(type, n));
   for (int i = 0; i < n; ++i) {
+    if (SHOULD_TICK) cli_progress_set(bar, i);
```

Comment:
Should that macro/constant be `CLI_SHOULD_TICK`?

Reply:
yeah.